addons/purchase_stock/models/purchase.py

```diff
@@ -285,12 +285,10 @@ def _create_or_update_picking(self):
                 if float_compare(line.product_qty, line.qty_invoiced, line.product_uom.rounding) == -1:
                     # If the quantity is now below the invoiced quantity, create an activity on the vendor bill
                     # inviting the user to create a refund.
-                    activity = self.env['mail.activity'].sudo().create({
-                        'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
-                        'note': _('The quantities on your purchase order indicate less than billed. You should ask for a refund. '),
-                        'res_id': line.invoice_lines[0].invoice_id.id,
-                        'res_model_id': self.env.ref('account.model_account_invoice').id,
-                    })
+                    activity = line.invoice_lines[0].invoice_id.activity_schedule(
+                        'mail.mail_activity_data_todo',
+                        note=_('The quantities on your purchase order indicate less than billed. You should ask for a refund.'))
+                    # VFE NOTE: isn't there a cleaner way to update activity deadline?
```

Comment:
What is this comment ?

Reply:
It is about the _onchange_activity_type_id() called just after. As I understood it, it is called to update activity deadline depending on the activity type, but it could maybe be removed now.